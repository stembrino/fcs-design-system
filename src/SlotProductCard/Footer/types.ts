export interface FooterProps {
  btnColor: string;
  onClick?: (event: React.MouseEvent<HTMLDivElement>) => void;
}
