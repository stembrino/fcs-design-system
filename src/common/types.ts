export interface CSSProperties extends React.CSSProperties {
  "--focus-color"?: string;
  "--button-color"?: string;
  "--border-color"?: string;
}
