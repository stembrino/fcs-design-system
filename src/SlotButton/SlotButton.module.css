:root {
  --label-overflow-padding: 16px;
}

div {
  box-sizing: border-box;
}

.base {
  --button-color: #304ffe;
  cursor: pointer;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  color: var(--slot-color-white);
  background-color: var(--button-color);
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  border-radius: 5px;
  padding-inline: var(--label-overflow-padding);
  min-width: 80px;
}

.base.rounded {
  border-radius: 100px;
}

.base.primary:hover {
  filter: brightness(85%);
}

.base.secondary {
  color: var(--button-color);
  background-color: transparent;
  border: 2px solid var(--button-color);
}

.base.secondary:hover {
  filter: grayscale(50%);
} 

.base.small {
  height: 32px;
}

.base.medium {
  height: 40px;
}

.base.large {
  height: 48px;
}

.label {
  margin-inline: var(--label-overflow-padding);
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}
