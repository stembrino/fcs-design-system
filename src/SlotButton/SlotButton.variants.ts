import { cva } from "class-variance-authority";
import styles from "./SlotButton.module.css";

export const variants = cva(styles.base, {
  variants: {
    theme: {
      primary: styles.primary,
      secondary: styles.secondary,
    },
    size: {
      small: styles.small,
      medium: styles.medium,
      large: styles.large,
    },
  },
  compoundVariants: [{ theme: "primary", size: "medium", className: styles.primaryMedium }],
  defaultVariants: {
    theme: "primary",
    size: "medium",
  },
});
