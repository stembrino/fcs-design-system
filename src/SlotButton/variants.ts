import { cva } from "class-variance-authority";
import styles from "./styles.module.css";

export const variants = cva(styles.base, {
  variants: {
    variant: {
      primary: styles.primary,
      secondary: styles.secondary,
    },
    size: {
      small: styles.small,
      medium: styles.medium,
      large: styles.large,
    },
  },
  compoundVariants: [{ variant: "primary", size: "medium" }],
  defaultVariants: {
    variant: "primary",
    size: "medium",
  },
});
