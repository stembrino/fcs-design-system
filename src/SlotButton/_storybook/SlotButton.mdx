import { Canvas, Meta, Controls } from '@storybook/blocks';
import * as SlotButton from "./SlotButton.stories";

<Meta of={SlotButton}  />

  # SlotButton

<Canvas of={SlotButton.Primary} />

<Controls />

