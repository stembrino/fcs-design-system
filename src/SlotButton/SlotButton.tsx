import React from "react";
import { SlotButtonProps } from "./types";
import styles from "./styles.module.css";
import { variants } from "./variants";
import { classNames } from "../common/util";
import { CSSProperties } from "../common/types";

const DefaultButton: React.FC<SlotButtonProps> = ({
  label,
  rounded = false,
  size = "medium",
  width = "auto",
  variant = "primary",
  color = "#304ffe",
  ariaLabel = "Button",
  focusColor = "#1c1b1b",
  onClick,
}) => {
  const roundedClass = rounded ? styles.rounded : "";
  const cssProperties: CSSProperties = {
    "--button-color": color,
    "--focus-color": focusColor,
    width,
  };

  return (
    <div
      style={{ ...cssProperties }}
      className={variants({ size, variant, className: classNames(roundedClass) })}
      role="button"
      onClick={onClick}
      aria-label={ariaLabel}
      tabIndex={0}
    >
      <span className={styles.label}>{label}</span>
    </div>
  );
};

export default DefaultButton;
