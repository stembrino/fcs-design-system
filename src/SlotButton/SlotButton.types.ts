import { type VariantProps } from "class-variance-authority";
import { variants } from "./SlotButton.variants";

export interface SlotButtonProps extends VariantProps<typeof variants> {
  label: string;
  rounded?: boolean;
  width?: string;
  color?: string;
  ariaLabel?: string;
  focusColor?: string;
  onClick: (event: React.MouseEvent<HTMLDivElement>) => void;
}
