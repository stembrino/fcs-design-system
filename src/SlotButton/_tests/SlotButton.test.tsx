import React from "react";
import { render, screen } from "@testing-library/react";

import SlotButton from "..";
import { SlotButtonProps } from "../SlotButton.types";

describe("TestComponent", () => {
  const renderComponent = ({ label = "test" }: Partial<SlotButtonProps>) => {
    const onClickMock = jest.fn();
    render(<SlotButton label={label} onClick={onClickMock} />);

    return {
      onClickMock,
    };
  };

  it("should render heading text correctly", () => {
    renderComponent({});

    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  it("should onClick be called when click", () => {
    const { onClickMock } = renderComponent({});

    expect(onClickMock).toBeCalledTimes(1);
  });
});
