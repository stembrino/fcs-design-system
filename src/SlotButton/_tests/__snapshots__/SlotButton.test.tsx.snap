// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent should snapshot match 1`] = `
<div>
  <div
    aria-label="Button"
    class="base primary medium"
    role="button"
    style="--button-color: #304ffe; --focus-color: #1c1b1b; width: auto;"
    tabindex="0"
  >
    <span
      class="label"
    >
      test
    </span>
  </div>
</div>
`;
